Update tip #22 'Overriding self with a weak reference' to Swift 4.2

juliancadi opened the pull request:
Using optional binding to upgrade self from a weak to strong reference. Now available in Swift 4.2

juliancadi committed:
Update tip #22 'Overriding self with a weak reference' to Swift 4.2

Using optional binding to upgrade self from a weak to strong reference. Now available in Swift 4.2

JohnSundell commented on README.md:
````diff
@@ -1919,6 +1919,21 @@ To stop the playground from executing, simply call `PlaygroundPage.current.finis
+
+```swift
+dataLoader.loadData(from: url) { [weak self] result in
+    guard let self = self else { return }
````
Please follow the same code style as in the previous tip (that is, put the `return` on a new line) 😇

JohnSundell commented on README.md:
````diff
@@ -1919,6 +1919,21 @@ To stop the playground from executing, simply call `PlaygroundPage.current.finis
+    ...
+```
+
+Accepted [proposal](https://github.com/apple/swift-evolution/blob/master/proposals/0079-upgrade-self-from-weak-to-strong.md) now implemented in Swift 4.2
````
I don't think this is relevant for this repo. Can you remove it?

juliancadi replied:
Sure 😉

JohnSundell reviewed (commented):
Thanks for this @juliancadi 🙌 Two changes I'd like you to make inline, after that I'll merge this in 👍

juliancadi committed:
Adjust tip format

- Consistent Code Style (return in new line).
- Remove irrelevant content about Swift proposal.

JohnSundell reviewed (approved):
Awesome, thanks @juliancadi! 🎉

juliancadi commented:
Thanks to you for such a great work with these tips! They are incredibly useful while coding :)
